File: app/src/main/java/com/lazysoul/kotlinwithandroid/injection/scopes/ActivityScope.java
package com.lazysoul.kotlinwithandroid.injection.scopes;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

/**
 * Created by Lazysoul on 06/01/2018.
 */

@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ActivityScope {
}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/injection/scopes/ActivityScope.kt
package com.lazysoul.kotlinwithandroid.injection.scopes

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

import javax.inject.Scope

/**
 * Created by Lazysoul on 2017. 7. 17..
 */

@Scope
@Retention(RetentionPolicy.RUNTIME)
annotation class ActivityScope
